### PR TITLE
gcp - big query job actions

### PR DESCRIPTION
**Pull request by alexkarpitski**
`Cancel` BigQuery Jobs.

### CODE REVIEW

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py [hunk 1]
```diff
@@ -56,6 +60,82 @@ def augment(self, resources):
         return results
 
 
+@DataSet.action_registry.register('delete')
+class DataSetActionDelete(MethodAction):
+    """The action is used for bigquery datasets delete.
+    GCP resource is https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets.
```
Comment (evhenyshaliov) [inline tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: Please delete the row

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py [hunk 2]
```diff
@@ -56,6 +60,82 @@ def augment(self, resources):
...
+            'projectId': project_id,
+            'datasetId': dataset_id,
+            'deleteContents': True
+        }
+
+
+@DataSet.action_registry.register('update-table-expiration')
+class DataSetActionPatch(MethodAction):
+    """The action is used for bigquery datasets defaultTableExpirationMs patch.
+    GCP resource is https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets.
```
Comment (evhenyshaliov) [inline tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: The row with resource link should be deleted when action link added

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py [hunk 3]
```diff
@@ -56,6 +60,82 @@ def augment(self, resources):
...
+        project_id, dataset_id = self.path_param_re.match(
+            resource['selfLink']).groups()
+        return {
+            'projectId': project_id,
+            'datasetId': dataset_id,
+            'deleteContents': True
+        }
+
+
+@DataSet.action_registry.register('update-table-expiration')
```
Comment (evhenyshaliov) [inline tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: The name of action should be replaced to set

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py [hunk 4]
```diff
@@ -56,6 +60,82 @@ def augment(self, resources):
...
+    .. code-block:: yaml
+        policies:
+          - name: gcp-big-dataset-update-table-expiration
+            resource: gcp.bq-dataset
+            filters:
+              - type: value
+                key: id
+                value: project_id:dataset_id
+            actions:
+              - type: update-table-expiration
```
Comment (evhenyshaliov) [inline tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: set

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py [hunk 5]
```diff
@@ -56,6 +60,82 @@ def augment(self, resources):
...
+                tableExpirationMs: 7200000
+    """
+
+    schema = type_schema(
+        'update-table-expiration',
+        **{
+            'type': {'enum': ['update-table-expiration']},
```
Comment (evhenyshaliov) [inline tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: The row could be deleted

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py [hunk 6]
```diff
@@ -80,6 +160,30 @@ def get(client, event):
             })
 
 
+@BigQueryJob.action_registry.register('cancel')
+class BigQueryJobStop(MethodAction):
+    """The action is used for bigquery jobs cancel.
+    GCP resource is https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs.
```
Comment (evhenyshaliov) [inline tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: The row with resource link should be deleted

File: tools/c7n_gcp/tests/test_bigquery.py [hunk 7]
```diff
@@ -31,6 +32,64 @@ def test_query(self):
...
+                           'value': dataset_id}],
+                 actions=[{'type': 'delete'}]),
+            session_factory=session_factory)
+        resources = policy.run()
+        self.assertEqual(resources[0]['id'], dataset_id)
+
+        if self.recording:
+            sleep(1)
+
+        policy = self.load_policy(base_policy, session_factory=session_factory)
```
Comment (evhenyshaliov) [inline tools/c7n_gcp/tests/test_bigquery.py]: Here should be implementation without policy loading. Use client based approach please

File: tools/c7n_gcp/tests/test_bigquery.py [hunk 8]
```diff
@@ -31,6 +32,64 @@ def test_query(self):
...
+                     'tableExpirationMs': 7200000
+                 }]),
+            session_factory=session_factory)
+        resources = policy.run()
+        self.assertEqual(resources[0]['id'], dataset_id)
+
+        if self.recording:
+            sleep(1)
+
+        policy = self.load_policy(base_policy, session_factory=session_factory)
```
Comment (evhenyshaliov) [inline tools/c7n_gcp/tests/test_bigquery.py]: Here should be implementation without policy loading. Use client based approach please

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py (hunk 1 above)
Comment (Dkhanas) [reply tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: Done

Comment (Dkhanas) [reply tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: Done

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py (hunk 2 above)
Comment (Dkhanas) [reply tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: Done

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py (hunk 3 above)
Comment (Dkhanas) [reply tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: Done

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py (hunk 4 above)
Comment (Dkhanas) [reply tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: Done

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py (hunk 5 above)
Comment (Dkhanas) [reply tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: Done

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py (hunk 6 above)
Comment (Dkhanas) [reply tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: Done

File: tools/c7n_gcp/tests/test_bigquery.py (hunk 7 above)
Comment (Dkhanas) [reply tools/c7n_gcp/tests/test_bigquery.py]: Done

File: tools/c7n_gcp/tests/test_bigquery.py (hunk 8 above)
Comment (Dkhanas) [reply tools/c7n_gcp/tests/test_bigquery.py]: Done

**Comment by aleonidex** [comment]
Not mandatory for Phase 1 and 2 at the moment, I'm closing it. Can be re-opened later if time permits.

File: docs/source/gcp/examples/bigquery-job-cancel.rst [hunk 9]
```diff
@@ -0,0 +1,16 @@
+BigQuery - Cancel job filtered job
```
Comment (evhenyshaliov) [inline docs/source/gcp/examples/bigquery-job-cancel.rst]: Update title

File: docs/source/gcp/examples/bigquery-job-cancel.rst [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+================================
+
+The example allows to delete filtered job.
+
+.. code-block:: yaml
+
+    policies:
+      - name: gcp-big-jobs-cancel
+        resource: gcp.bq-job
+        filters:
```
Comment (evhenyshaliov) [inline docs/source/gcp/examples/bigquery-job-cancel.rst]: Update filters. It should work with more than 1 entity

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py [hunk 11]
```diff
@@ -55,6 +59,80 @@ def augment(self, resources):
...
+              - type: value
+                key: id
+                value: project_id:dataset_id
+            actions:
+              - type: set-table-expiration
+                tableExpirationMs: 7200000
+    """
+
+    schema = type_schema(
+        'set-table-expiration',
```
Comment (evhenyshaliov) [inline tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: Replace by set

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py [hunk 12]
```diff
@@ -55,6 +59,80 @@ def augment(self, resources):
...
+        **{
+            'type': {'enum': ['set-table-expiration']},
```
Comment (evhenyshaliov) [inline tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: This row should be removed

File: tools/c7n_gcp/c7n_gcp/resources/bigquery.py [hunk 13]
```diff
@@ -55,6 +59,80 @@ def augment(self, resources):
...
+            'tableExpirationMs': {
```
Comment (evhenyshaliov) [inline tools/c7n_gcp/c7n_gcp/resources/bigquery.py]: Please add other params that can have use cases

**Comment by Dkhanas** [comment]
code moved https://github.com/mediapills/cloud-custodian/pull/125

**Pull request closed without merging**